nixos/modules/services/games/mindustry.nix

```diff
@@ -0,0 +1,115 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.mindustry;
+  name = "Mindustry";
+
+  serverConfig = {
+    name = cfg.game-name;
+    port = (toString cfg.port);
+  } // cfg.extraConfig;
+  serverConfigCommands = (attrsets.mapAttrsToList (k: v:
+    "config ${k} ${if builtins.isBool v then boolToString v else toString v}")
+    serverConfig);
+  serverCommands = cfg.extraCommands ++ serverConfigCommands
+    ++ [ "host ${cfg.game-map} ${cfg.game-mode}" ];
+  stateDir = "/var/lib/${cfg.stateDirName}";
+in {
+  options = {
+    services.mindustry = {
+      enable = mkEnableOption name;
+      package = mkOption {
```

Comment:
Are there any variants of this package that make this option especially convenient?

Reply:
Yes, there's the version whitelist override, which is a compile-time option